### PR TITLE
8341513: Remove the unused thread_type field from OSThread

### PR DESCRIPTION
**Pull request by offamitkumar**
This trivial PR cleans up ` thread_type` field. 

Testing: 
Build + Tier1 test (Fastdebug vm) are __clean__, no regression seen on linux/s390x.

I have tested on local system Mac-M1 but it failed due to this issue: [JDK-8340341](https://bugs.openjdk.org/browse/JDK-8340341)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8341513](https://bugs.openjdk.org/browse/JDK-8341513): Remove the unused thread_type field from OSThread (**Enhancement** - P4)

### Reviewers
 * [Stefan Karlsson](https://openjdk.org/census#stefank) (@stefank - **Reviewer**)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21532/head:pull/21532` \
`$ git checkout pull/21532`

Update a local copy of the PR: \
`$ git checkout pull/21532` \
`$ git pull https://git.openjdk.org/jdk.git pull/21532/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21532`

View PR using the GUI difftool: \
`$ git pr show -t 21532`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21532.diff">https://git.openjdk.org/jdk/pull/21532.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21532#issuecomment-2418491936)

### CODE REVIEW

**Review by stefank — APPROVED** [review]
Looks good to me. @dholmes-ora did some digging into the history of this, and it would be good to get his review on this change as well.

File: src/hotspot/os/aix/osThread_aix.cpp [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*
- * Copyright (c) 1999, 2023, Oracle and/or its affiliates. All rights reserved.
- * Copyright (c) 2012, 2015 SAP SE. All rights reserved.
+ * Copyright (c) 1999, 2024, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2012, 2024 SAP SE. All rights reserved.
```
Comment (dholmes-ora) [inline src/hotspot/os/aix/osThread_aix.cpp:3]: @offamitkumar  do you work for SAP? If not then please do not modify their copyright notice. Thanks for updating the Oracle one's as I requested.

**Review by dholmes-ora — COMMENTED** [review]
Functional changes are good.

Please run GHA on your branch so that all builds are tested, and successful.

Thanks

**Comment by offamitkumar** [comment]
>Please run GHA on your branch so that all builds are tested, and successful.

GHA are enabled on my repo from initial commit.

**Review by stefank — APPROVED** [review]

**Review by dholmes-ora — APPROVED** [review]
Your GHA is still running and appeared to start about 3 hours ago.

Your comments in the PR description indicated very limited testing. Please edit the comment to reflect successful testing. Thanks

**Comment by offamitkumar** [comment]
Only MacOS jobs are stuck: 
> -> offamitkumar/jdk/.github/workflows/test.yml@refs/heads/thread_type_rm (5c5dead5d7129fbdbbf24f298a597dbd5657a336)
Waiting for a runner to pick up this job...


So I will wait before integration. Because on my local system the build failed with the issue I mentioned in the description.

**Comment by offamitkumar** [comment]
GHA result are fine. Thanks @stefank @dholmes-ora for the reviews. 

/integrate

**Pull request closed without merging**